### PR TITLE
Keep tensors alive for Dynamo pointer reconstruction

### PR DESCRIPTION
**Pull request by jansel**
Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.16.0) (oldest at bottom):
* __->__ #185467

When Dynamo graph-breaks after creating a tensor in the captured graph, the resume bytecode may need to reconstruct raw pointer-like values such as a.untyped_storage()._cdata. Before this change, _cdata was represented as a generic GetAttrVariable, so the resumed bytecode loaded graph_out[0].untyped_storage()._cdata and then deleted graph_out before calling the remaining Python function. That left torch._C._storage_Use_Count with a dangling StorageImpl pointer and produced a bogus count.

Teach UntypedStorageVariable to return a dedicated StorageCDataVariable for _cdata, and reconstruct it by stashing the owning tensor in a temporary local before exposing the raw integer. Apply the same keepalive reconstruction to DataPtrVariable, which represents the same class of raw pointer value across graph breaks.

Fixes #156059

Generated by my agent

Test Plan:
- python test/dynamo/test_misc.py MiscTests.test_storage_cdata_use_count_graph_break_lifetime
- lintrunner -a
- python - <<'PY'
  import torch

  @torch._dynamo.optimize('eager')
  def get_ref():
      a = torch.randn(10)
      ref = torch._C._storage_Use_Count(a.untyped_storage()._cdata)
      print('Reference count:', ref)

  get_ref()
  PY

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @jataylo @azahed98

### CODE REVIEW

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says I rebased the stack on current main and added data_ptr graph-break lifetime regression coverage, addressing the review-bot suggestion. The previous unclassified CI failures were SIGKILLs in unrelated large serialization tests; CI has restarted on the new head, and I added the topic: not user facing label.

**Comment by jansel** [comment]
My agent says CI had one inductor OSDC failure before tests started due to a runner custom-container error, which matches the active runner SEV. I reran the failed job and will check the restarted CI after it completes.

**Comment by jansel** [comment]
My agent says the only remaining CI signal was the same inductor OSDC shard. Its previous rerun got through setup and several test shards before failing without a product-code error in the exposed failed log, so I reran the failed job again; the replacement job is queued at https://github.com/pytorch/pytorch/actions/runs/26612447609/job/79129274791.

**Comment by jansel** [comment]
My agent says I rebased the PR onto current main, fixed a current-main lint failure in the new Dynamo annotation by using InstructionTranslatorBase, and resubmitted the stack. Local targeted tests and lintrunner pass; CI has restarted on the new head f88069ffe5216f05ad9b57a49e5deba96a231cfd.

**Comment by mlazos** [comment]
@claude Does this increase the lifetime of these tensors, if so by how much?

**Comment by jansel** [comment]
My agent says the lifetime increase is limited to tensors whose raw pointer is reconstructed across a graph break, and those tensors are held only until the resume frame exits. That bounded frame-local keepalive is the intended fix: it keeps the owner alive long enough that the raw pointer cannot dangle after graph_out is deleted.

**Comment by jansel** [comment]
My agent says I rebased this PR onto current main and resolved the Dynamo attribute-dispatch conflict by registering _cdata through tp_getset while preserving the frame-local owner keepalive. Both focused pointer-lifetime tests pass before and after lintrunner, the direct repro now matches eager at 2, and CI has restarted on the updated head.

**Comment by jansel** [comment]
@claude review these changes

**Comment by jansel** [comment]
My agent says this PR is being closed as part of a cleanup of open agentic PRs to focus review bandwidth on a smaller set; the linked issue remains open.
